src/ui/wxWidgets/SystemTray.cpp

```diff
@@ -302,6 +302,7 @@ void SystemTray::ShowSetDatabaseIdDialog()
 void SystemTray::ProcessSysTrayMenuItem(int itemId)
 {
   switch(itemId) {
+    case ID_SYSTRAY_UNLOCK:
     case ID_SYSTRAY_RESTORE:
```

Comment:
In case of Restore, the database should be unlocked and shown to the user again.